### PR TITLE
test_physfs: parse arguments + add tree command

### PR DESCRIPTION
**Pull request by madebr**
- CMake will drop support for cmake scripts < 3.5, so bump it to 3.5
- Use rpath for the executables, such that `test_physfs` uses the newly built libphysfs.so library, instead of the system library.
These paths are removed when they are installed.
- Use the alias'ed targets
- Treat arguments as interactive commands.
This allows to run commands from the shell without requiring to put them in a file.
This was useful while developing in the IDE, not needing to manually enter something.
This allows e.g. `./test_physfs "mount /tmp/data_tree /tt 0" "ls /tt" "quit"`
- Add a `tree` command to `test_physfs`. This works more or less similar to the unix tree command: non-accessible files are ignored + symlinks are not entered.

The following comands:
```sh
cat <<EOF >tree.sh
#!/bin/sh

set -e

PARENT=\$(pwd)/data_tree

function create_tree() {
  local i
  if test \$1 -ne 0; then
    for i in \$(seq \$2); do
      mkdir -p "dir\$i"
      pushd "dir\$i" >/dev/null
      create_tree \$(expr \$1 - 1 ) \$2
      popd >/dev/null
      if test \$i -eq 2; then
        chmod 0 dir\$i
      fi
    done
    ln -s "\$PARENT" dir\$(expr \$2 + 1)
  fi
  for i in \$(seq \$2); do
    echo "file\$1" >"file\$i"
  done
}

sudo rm -rf "\$PARENT"
mkdir "\$PARENT"
pushd "\$PARENT" >/dev/null
create_tree 3 3
popd >/dev/null
EOF

sh tree.sh
./test_physfs "mount $PWD/data_tree /tt 0" "tree /tt" "quit"
```

will output:
```
...
Enter commands. Enter "help" for instructions.
Successful.
/tt
├── dir1
│   ├── dir1
│   │   ├── dir1
│   │   │   ├── file1
│   │   │   ├── file2
│   │   │   └── file3
│   │   ├── dir2 [Failure. reason: permission denied]
│   │   ├── dir3
│   │   │   ├── file1
│   │   │   ├── file2
│   │   │   └── file3
│   │   ├── file1
│   │   ├── file2
│   │   └── file3
│   ├── dir2 [Failure. reason: permission denied]
│   ├── dir3
│   │   ├── dir1
│   │   │   ├── file1
│   │   │   ├── file2
│   │   │   └── file3
│   │   ├── dir2 [Failure. reason: permission denied]
│   │   ├── dir3
│   │   │   ├── file1
│   │   │   ├── file2
│   │   │   └── file3
│   │   ├── file1
│   │   ├── file2
│   │   └── file3
│   ├── file1
│   ├── file2
│   └── file3
├── dir2 [Failure. reason: permission denied]
├── dir3
│   ├── dir1
│   │   ├── dir1
│   │   │   ├── file1
│   │   │   ├── file2
│   │   │   └── file3
│   │   ├── dir2 [Failure. reason: permission denied]
│   │   ├── dir3
│   │   │   ├── file1
│   │   │   ├── file2
│   │   │   └── file3
│   │   ├── file1
│   │   ├── file2
│   │   └── file3
│   ├── dir2 [Failure. reason: permission denied]
│   ├── dir3
│   │   ├── dir1
│   │   │   ├── file1
│   │   │   ├── file2
│   │   │   └── file3
│   │   ├── dir2 [Failure. reason: permission denied]
│   │   ├── dir3
│   │   │   ├── file1
│   │   │   ├── file2
│   │   │   └── file3
│   │   ├── file1
│   │   ├── file2
│   │   └── file3
│   ├── file1
│   ├── file2
│   └── file3
├── file1
├── file2
└── file3

21 directories, 45 files
```

(I don't know why no symbolic links show up)

### CODE REVIEW

File: test/test_physfs.c [hunk 1]
```diff
@@ -380,6 +380,86 @@ static int cmd_enumerate(char *args)
     return 1;
 } /* cmd_enumerate */
 
+#define STR_BOX_VERTICAL_RIGHT  "\xe2\x94\x9c"
+#define STR_BOX_VERTICAL        "\xe2\x94\x82"
+#define STR_BOX_HORIZONTAL      "\xe2\x94\x80"
+#define STR_BOX_UP_RIGHT        "\xe2\x94\x94"
+#define STR_NBSP                "\xc2\xa0"
```
Comment (madebr) [inline test/test_physfs.c:387]: These are utf-8 encodings of ├, │, └ and ─ + a space.

File: test/test_physfs.c [hunk 2]
```diff
@@ -1631,7 +1717,7 @@ int main(int argc, char **argv)
         fflush(stdout);
         if (buf != NULL)
             free(buf);
-    } while (rc);
+    };
```
Comment (madebr) [inline test/test_physfs.c:1720]:
```suggestion
    }
```

**Comment by icculus** [comment]
> (I don't know why no symbolic links show up)

PhysicsFS explicitly ignores symlinks by default. Try this:

```bash
./test_physfs "permitsymlinks 1" "mount $PWD/data_tree /tt 0" "tree /tt" "quit"
```

**Comment by madebr** [comment]
Yup, that works.
```
/tt
├── dir1
│   ├── dir1
│   │   ├── dir1
│   │   │   ├── file1
│   │   │   ├── file2
│   │   │   └── file3
│   │   ├── dir2 [Failure. reason: permission denied]
│   │   ├── dir3
│   │   │   ├── file1
│   │   │   ├── file2
│   │   │   └── file3
│   │   ├── dir4 [symbolic link]
│   │   ├── file1
│   │   ├── file2
│   │   └── file3
│   ├── dir2 [Failure. reason: permission denied]
│   ├── dir3
│   │   ├── dir1
│   │   │   ├── file1
│   │   │   ├── file2
│   │   │   └── file3
│   │   ├── dir2 [Failure. reason: permission denied]
│   │   ├── dir3
│   │   │   ├── file1
│   │   │   ├── file2
│   │   │   └── file3
│   │   ├── dir4 [symbolic link]
│   │   ├── file1
│   │   ├── file2
│   │   └── file3
│   ├── dir4 [symbolic link]
│   ├── file1
│   ├── file2
│   └── file3
├── dir2 [Failure. reason: permission denied]
├── dir3
│   ├── dir1
│   │   ├── dir1
│   │   │   ├── file1
│   │   │   ├── file2
│   │   │   └── file3
│   │   ├── dir2 [Failure. reason: permission denied]
│   │   ├── dir3
│   │   │   ├── file1
│   │   │   ├── file2
│   │   │   └── file3
│   │   ├── dir4 [symbolic link]
│   │   ├── file1
│   │   ├── file2
│   │   └── file3
│   ├── dir2 [Failure. reason: permission denied]
│   ├── dir3
│   │   ├── dir1
│   │   │   ├── file1
│   │   │   ├── file2
│   │   │   └── file3
│   │   ├── dir2 [Failure. reason: permission denied]
│   │   ├── dir3
│   │   │   ├── file1
│   │   │   ├── file2
│   │   │   └── file3
│   │   ├── dir4 [symbolic link]
│   │   ├── file1
│   │   ├── file2
│   │   └── file3
│   ├── dir4 [symbolic link]
│   ├── file1
│   ├── file2
│   └── file3
├── dir4 [symbolic link]
├── file1
├── file2
└── file3
```

**Pull request merged**